### PR TITLE
fix `invalid-states`

### PR DESCRIPTION
**Pull request by anthonygalea**

### CODE REVIEW

File: src/automata/fsm.cljc [hunk 1]
```diff
@@ -50,9 +50,10 @@
   [rules]
   (let [valid-states (set (keys rules))]
     (-> (into #{}
-              (comp (mapcat (comp ::to val))
+              (comp (mapcat val)
+                    (map ::to)
                     (distinct)
-                    (remove #(contains? valid-states %)))
+                    (remove valid-states))
```
Comment (pyr) [inline src/automata/fsm.cljc]: I'm a bit against directly relying on the fact that sets implement **IFn**, let's go with  `(partial contains? valid-states)` to make it easier on future readers

File: src/automata/fsm.cljc [hunk 2]
```diff
@@ -50,9 +50,10 @@
   [rules]
   (let [valid-states (set (keys rules))]
     (-> (into #{}
-              (comp (mapcat (comp ::to val))
+              (comp (mapcat val)
+                    (map ::to)
```
Comment (pyr) [inline src/automata/fsm.cljc:54]: :+1:

**Review by pyr — COMMENTED** [review]
Thanks @anthonygalea !

File: src/automata/fsm.cljc (hunk 1 above)
Comment (anthonygalea) [reply src/automata/fsm.cljc]: ok, I switched to `(partial contains? valid-states)` in 9a8b0b9

**Review by pyr — APPROVED** [review]

**Pull request merged**